PostProcess/ScreenSpaceAmbientOcclusion/src/ScreenSpaceAmbientOcclusion.cpp

```diff
@@ -150,16 +150,26 @@ void ScreenSpaceAmbientOcclusion::PrepareResources(IRenderDevice* pDevice, PostF
         m_Resources.Insert(RESOURCE_IDENTIFIER_OCCLUSION_RESOLVED, Device.CreateTexture(Desc));
     }
 
+    auto ClearRenderTarget = [&](ITexture* pRenderTexture, float ClearColor[]) {
+        auto pRTV = pRenderTexture->GetDefaultView(TEXTURE_VIEW_RENDER_TARGET);
+        pDeviceContext->SetRenderTargets(1, &pRTV, nullptr, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+        pDeviceContext->ClearRenderTarget(pRTV, ClearColor, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+        pDeviceContext->SetRenderTargets(0, nullptr, nullptr, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+    };
```

Comment:
Here and in other places: the targets are set and cleared in the prepare method, which may be unexpected from the application side.
It is preferred to clear the target before the first use, if needed.